[pipelining] Improve schedule csv loading

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142009

Add small changes based on feedback from Less when testing out https://github.com/pytorch/torchtitan/pull/707
- expose `validate_schedule` as a function
- handle spaces around actions in csv file
- add error arrow to `_format_pipeline_schedule()` to better show where the step errored


cc @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

H-Huang committed:
[pipelining] Improve schedule csv loading

[ghstack-poisoned]

lessw2020 reviewed (approved):
lgtm, and thanks for modularizing/updating this functionality.

H-Huang committed:
Update on "[pipelining] Improve schedule csv loading"


Add small changes based on feedback from Less when testing out https://github.com/pytorch/torchtitan/pull/707
- expose `validate_schedule` as a function
- handle spaces around actions in csv file
- add error arrow to `_format_pipeline_schedule()` to better show where the step errored


cc awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge
